src/miniscript/iter.rs

```diff
@@ -76,6 +76,38 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Miniscript<Pk, Ctx> {
         }
     }
 
+    /// Returns child node with given index, if any
+    pub fn get_nth_child(&self, n: usize) -> Option<&Miniscript<Pk, Ctx>> {
+        match (n, &self.node) {
+            (0, Terminal::Alt(ref node))
+            | (0, Terminal::Swap(ref node))
+            | (0, Terminal::Check(ref node))
+            | (0, Terminal::DupIf(ref node))
+            | (0, Terminal::Verify(ref node))
+            | (0, Terminal::NonZero(ref node))
+            | (0, Terminal::ZeroNotEqual(ref node))
+            | (0, Terminal::AndV(ref node, _))
+            | (0, Terminal::AndB(ref node, _))
+            | (0, Terminal::OrB(ref node, _))
+            | (0, Terminal::OrD(ref node, _))
+            | (0, Terminal::OrC(ref node, _))
+            | (0, Terminal::OrI(ref node, _))
+            | (1, Terminal::AndV(_, ref node))
+            | (1, Terminal::AndB(_, ref node))
+            | (1, Terminal::OrB(_, ref node))
+            | (1, Terminal::OrD(_, ref node))
+            | (1, Terminal::OrC(_, ref node))
+            | (1, Terminal::OrI(_, ref node))
+            | (0, Terminal::AndOr(ref node, _, _))
+            | (1, Terminal::AndOr(_, ref node, _))
+            | (2, Terminal::AndOr(_, _, ref node)) => Some(node),
+
+            (n, Terminal::Thresh(_, ref node_vec)) => node_vec.get(n).map(Deref::deref),
```

Comment:
A minor nit - I think it's more idiomatic (and a bit shorter) to write `|x| **x` rather than `Deref::deref`

Reply:
Addressed this with latest force-push. But it seems to me `|x| &**x` looks really stranger than `Deref` :)

Reply:
why the `&`?

Reply:
B/c it does not compile without: we return Option<&>, not just option

Reply:
I think you meant `|x| *x` rather than `|x| &**x`

Reply:
It really takes `&**` to get from `&Arc<Miniscript>` (which we need to double-deref first with **) to `&Miniscript`

Reply:
> I think you meant |x| *x rather than |x| &**x

Nope, it does not compile that way.

`*x = *&Arc = Arc`

Reply:
Proof:

```console
error[E0308]: match arms have incompatible types
   --> src/miniscript/iter.rs:81:9
    |
81  | /         match (n, &self.node) {
82  | |             (0, &Terminal::Alt(ref node))
83  | |             | (0, &Terminal::Swap(ref node))
84  | |             | (0, &Terminal::Check(ref node))
...   |
107 | |             _ => None,
108 | |         }
    | |_________^ expected reference, found struct `std::sync::Arc`
    |
    = note: expected type `std::option::Option<&miniscript::Miniscript<Pk, Ctx>>`
               found type `std::option::Option<std::sync::Arc<miniscript::Miniscript<Pk, Ctx>>>`
note: match arm with an incompatible type
   --> src/miniscript/iter.rs:105:56
    |
105 |             (n, &Terminal::Thresh(_, ref node_vec)) => node_vec.get(n).map(|x| *x),
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Sorry, I forgot we had `Arc`s in there.